crates/types/src/blockchain/header.rs

```diff
@@ -112,26 +112,31 @@ impl BlockHeader {
     /// Set the block height for the header
     pub fn set_block_height(&mut self, height: BlockHeight) {
         self.consensus_mut().height = height;
+        self.recalculate_metadata();
```

Comment:
It looks like we have this method in a separate `impl` block for some reason. Both blocks for `BlockHeader` have the same (no) constraints. Can we combine the `impl` blocks?

Reply:
They are segregated by their business logic. I'm okay with joining them into one/two impl blocks, but since they are in different places in the file, I prefer to not shuffle them in this PR to minimize number of changes=)

Reply:
Okay. I just got confused where a method was just reading the PR diffs, but with an IDE it doesn't really affect anything.